components/arm/arm.go

```diff
@@ -143,55 +128,22 @@ func NamesFromRobot(r robot.Robot) []string {
 // CreateStatus creates a status from the arm. This will report calculated end effector positions even if the given
 // arm is perceived to be out of bounds.
 func CreateStatus(ctx context.Context, a Arm) (*pb.Status, error) {
-	jointPositions, err := a.JointPositions(ctx, nil)
+	jointPositions, err := a.CurrentInputs(ctx)
 	if err != nil {
 		return nil, err
 	}
 	model := a.ModelFrame()
 
 	var endPosition *v1.Pose
-	if endPose, err := motionplan.ComputeOOBPosition(model, jointPositions); err == nil {
+	if endPose, err := referenceframe.ComputeOOBPosition(model, jointPositions); err == nil {
```

Comment:
Just as an fyi - this entire function is being deprecated because `Status` messages are being deprecated https://github.com/viamrobotics/api/pull/518/files#diff-dc55009b957e10750e20d244efd3104b50bd1659e67a8d1917286a44b692bf0d, where should this logic live in the future? The arm client?

Reply:
If so we should just make that change now and here otherwise I feel like these functions will just be deleted wholesale without thought when the time comes, this is the only one doing fancy things other than board.

Reply:
This can all just go away when status messages do I think

Reply:
I agree glad to see it go away completely.

Reply:
Yeah I'm on board with just deleting the function when we need to

Reply:
Then how would you feel about just passing the apis to status without the checks right now? It makes development smoother because viam-server won't start spamming you with messages from the robot status stream if you've got bad limits in your kinematics file as you're developing, and it thinks the end position is actually OOB (An intern hit this last year)

Reply:
Would prefer to do that as a separate ticket if thats ok.  Was just intending on removing the dependency (and resolving the fallout from that) not rewriting the Status messages

Reply:
sure thing, I can also pick it up if you want.